common/iterator/BaseIterator.java

```diff
@@ -18,31 +18,28 @@
 
 package com.vaticle.typedb.core.common.iterator;
 
-import com.vaticle.typedb.common.collection.Either;
-
 import java.util.Iterator;
 
 class BaseIterator<T> extends AbstractFunctionalIterator<T> {
 
-    private final Either<FunctionalIterator<T>, Iterator<T>> iterator;
+    private final Iterator<T> iterator;
 
-    BaseIterator(Either<FunctionalIterator<T>, Iterator<T>> iterator) {
```

Comment:
profiling suggested that making the `BaseIterator` even a little bit more complex than it had to could have an impact on performance since it's one of the hottest code paths in the TypeDB! Here we simplify it here to remove an API that we never used: a `BaseIterator` wrapping a `Functionalterator`